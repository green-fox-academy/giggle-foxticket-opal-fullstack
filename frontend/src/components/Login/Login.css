html,
body {
  margin: 0;
  height: 100%;
}
*{
  font-family: Georgia, 'Times New Roman', Times, serif;
  font-size: 1em;

}

body {
  display: flex;
  align-items: center;
  justify-content: center;
}

.container {
  box-shadow: 0px 4px 5px 5px rgba(186, 186, 186, 1);
  width: 20em;
  padding: 2rem 1rem 0 1rem;
  text-align: center;
}
h1 {
  color: green;
  font-size: 5em;
  font-weight: 450;
  margin-left: 15px;
}

.login {
  background-color: green;
  color: white;
  padding: 5px;
  margin: 1em;
  padding: 5px 15px;
  border-radius: 3px;
  cursor: pointer;
}
.login:hover {
  background-color: lightgreen;
  color: white;
  padding: 5px;
  margin: 1em;
  padding: 5px 15px;
  border-radius: 3px;
}

.register {
  color: green;
  text-decoration: none;
  font-family: strong;
  margin-right: 8em;
}
.register:hover {
  color: lightgreen;

}
.register {
  color: green;
  text-decoration: none;
  font-family: strong;
  margin-right: 8em;
}

.isValid {
  visibility: hidden;
}

 .input-green {
  border: none;
  cursor: pointer;
  border-bottom: 2px solid green;
  margin: 0.5em;
}
 .input-green:focus {
  outline: none;
}
.input-red {
  border: none;
  cursor: pointer;
  border-bottom: 2px solid red;
  margin: 0.5em;

}
.input-red:focus {
 outline: none;
}

