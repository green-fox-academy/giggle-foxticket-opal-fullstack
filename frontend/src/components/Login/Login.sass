.login-body
  height: 600px
  font-family: Georgia, 'Times New Roman', Times, serif
  font-size: 1em
  display: flex
  flex-direction: column
  align-items: center
  justify-content: center

.login_title
  color: green
  font-size: 5em
  font-weight: 450

.login_container
  box-shadow: 0px 4px 5px 5px rgba(186, 186, 186, 1)
  width: 20em
  padding: 2rem 1rem 0 1rem
  text-align: center
  justify-content: center


.login_btn
  background-color: green
  color: white
  padding: 5px 15px
  border-radius: 3px
  cursor: pointer
  width: 55%
  margin-top: 10px
  margin-bottom: 10px


.login_btn:hover
  background-color: lightgreen
  color: white
  padding: 5px
  padding: 5px 15px
  border-radius: 3px

.login_register
  color: green
  text-decoration: none
  font-family: strong
  margin-right: 8em

.login_register:hover
  color: lightgreen

.isValid
  visibility: hidden

.notValid
  display: flex
  align-items: center
  justify-content: space-around

.input-green
  border: none
  cursor: pointer
  border-bottom: 2px solid green
  margin: 0.5em

.input-green:focus
  outline: none

.input-red
  border: none
  cursor: pointer
  border-bottom: 2px solid red
  margin: 0.5em

.input-red:focus
  outline: none

