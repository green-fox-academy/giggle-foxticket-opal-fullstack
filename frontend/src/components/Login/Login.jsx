import React, { useState } from 'react';
import { BrowserRouter, NavLink, Route } from 'react-router-dom';
import './Login.css';

import { FaExclamationTriangle } from 'react-icons/fa';

const Login = () => {
  const [name, setName] = useState('');
  const [password, setPassword] = useState('');
  const [isValid, setIsValid] = useState(true);

  const handleNameChange = e => {
    setName(e.target.value);
  };
  const handlePasswordChange = e => {
    setPassword(e.target.value);
  };

  const warningMsg = {
    color: isValid ? 'green' : 'red',
  };

  const postHandler = e => {
    e.preventDefault();
    if (name.length <= 3 || password.length <= 3) {
      setIsValid(false);
    }
    const user = { name, password };

    console.log(user);
    fetch('http://localhost:3000/api/session', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(user),
    });
  };

  return (
    <BrowserRouter>
      <Route>
        <h1>Foxticket</h1>
        <div className="container">
          <form onSubmit={postHandler}>
            <input
              className={'input-' + warningMsg.color}
              type="text"
              name="username"
              id="username"
              placeholder="Username"
              required
              onChange={handleNameChange}
            />
            <input
              className={'input-' + warningMsg.color}
              type="password"
              name="password"
              id="password"
              placeholder="Password"
              required
              onChange={handlePasswordChange}
            />
            <p style={warningMsg} className={` ${isValid ? 'isValid' : ''}`}>
              Username or Password is incorrect
              <FaExclamationTriangle color="red" size="1.5em" />
            </p>

            <i className="fas fa-exclamation-triangle"></i>

            <NavLink className="register" to="/register" exact>
              REGISTER
            </NavLink>
            <button className="login">Login</button>
          </form>
        </div>
      </Route>
    </BrowserRouter>
  );
};

export default Login;
