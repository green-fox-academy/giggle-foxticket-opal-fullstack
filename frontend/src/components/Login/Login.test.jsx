import React from 'react';
import Login from './Login';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';
test('first snapshot of Login', () => {
  const component = renderer.create(
    <MemoryRouter>
      <Login />
    </MemoryRouter>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
