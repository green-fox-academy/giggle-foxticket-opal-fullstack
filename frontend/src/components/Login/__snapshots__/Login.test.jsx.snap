// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`first snapshot of Login 1`] = `
<div
  className="login-body"
>
  <div
    className="login_title"
  >
    Foxticket
  </div>
  <div
    className="login_container"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        className="input-green"
        id="username"
        name="username"
        onChange={[Function]}
        placeholder="Username"
        required={true}
        type="text"
      />
      <input
        className="input-green"
        id="password"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
      />
      <p
        className=" isValid"
        style={
          Object {
            "color": "green",
          }
        }
      >
        Username or Password is incorrect
        <svg
          color="red"
          fill="currentColor"
          height="1.5em"
          size="1.5em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "red",
            }
          }
          viewBox="0 0 576 512"
          width="1.5em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
          />
        </svg>
      </p>
      <i
        className="fas fa-exclamation-triangle"
      />
      <a
        className="login_register"
        exact={true}
        href="/register"
        onClick={[Function]}
      >
        REGISTER
      </a>
      <button
        className="login_btn"
      >
        Login
      </button>
    </form>
  </div>
</div>
`;
